Refactor deep tests to separate file, fix deep chunk reads, ripmap reading

kdt3rd opened the pull request:
Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Refactor deep tests to separate file, fix deep chunk reads, ripmap reading

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

cary-ilm commented on src/lib/OpenEXRCore/unpack.c:
```diff
@@ -992,6 +995,186 @@ generic_unpack (exr_decode_pipeline_t* decode)
     return EXR_ERR_SUCCESS;
 }
 
+//static exr_result_t
```
Should this commented-out code be removed altogether?

cary-ilm commented on src/lib/OpenEXRCore/unpack.c:
```diff
@@ -1017,7 +1200,8 @@ internal_exr_match_decode (
 
-    /* TODO */
+    /* we don't decode deep for now, it is up to the user to do that (for efficiency) */
+    //if (isdeep) return &generic_unpack_deep;
```
Should this commented-out line be removed?

cary-ilm reviewed (approved):
LGTM, except for the commented-out code. OK to keep it temporarily if it's useful.

kdt3rd commented:
I was misinterpreting the data coming in, which is why I had disabled this thinking I would limit scope. But after talking to Peter today, I think I will uncomment it shortly and provide that again

kdt3rd committed:
Fix issues with reading deep files

- add safety check to zip
- fix piz decompressor
- add optimization for unpack
- fix pointer deep read
- reenable deep decoding

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
